File: arduino/Sensorkit/src/Dht11Sensor.cpp
#include "Dht11Sensor.h"
#define DHTTYPE DHT11 // DHT 11

Dht11Sensor::Dht11Sensor(int sensor) {
  sensorPin = sensor;
  dhtTemperature = 0.0;
  dhtHumidity = 0.0;
  dht = new DHT(sensorPin, DHTTYPE);
}
Dht11Sensor::~Dht11Sensor() {
  delete dht;
}

void Dht11Sensor::setup() {
  dht->begin();
  dhtLastMillis = millis();
}
float Dht11Sensor::temperature(){
  read();
  return dhtTemperature;
}
float Dht11Sensor::humidity(){
  read();
  return dhtHumidity;
}
void Dht11Sensor::read(){
  unsigned long dhtMillis = millis();
  if (dhtMillis - dhtLastMillis > 1000) {
    dhtTemperature = dht->readTemperature();
    dhtHumidity = dht->readHumidity();
    dhtLastMillis = millis();
  }
}

File: micropython/Sensorkit/Dht11Sensor.py
from machine import Pin
from time import ticks_ms
import dht;

# A sensor to measure the temperature and humidity of the environment.
# interfaces on the dht.DHT11 environment sensor.
# new values are measured every 1000 ms.
# last read values are provided in the meantime.
class Dht11Sensor():
  dhtLastMillis = 0
  dhtMillis = 0
  dhtMeasureMillis = 1000
  dhtTemperature = 0.0
  dhtHumidity = 0.0
  dhtErrors = 0
  def __init__(self, MEASURE):
    global sensor, dhtMillis, dhtTemperature, dhtHumidity
    sensor = dht.DHT11(Pin(MEASURE));
    self.dhtLastMillis = ticks_ms()
  def temperature(self):
    self.read()
    return self.dhtTemperature
  def humidity(self):
    self.read()
    return self.dhtHumidity
  def read(self):
    try:
      self.dhtMillis = ticks_ms()
      if self.dhtMillis - self.dhtLastMillis > self.dhtMeasureMillis:
        sensor.measure()
        self.dhtTemperature = sensor.temperature()
        self.dhtHumidity = sensor.humidity()
        self.dhtLastMillis = ticks_ms();
    except OSError as e:
      self.dhtErrors = self.dhtErrors + 1
      #print('Failed to read sensor.')
